sdk-extensions/autoconfigure/README.md

```diff
@@ -329,11 +329,12 @@ The following exporters are only available for the metric signal. See [exporters
 
 The [Prometheus](https://github.com/prometheus/docs/blob/master/content/docs/instrumenting/exposition_formats.md) exporter.
 
-| System property                  | Environment variable             | Description                                                                        |
-|----------------------------------|----------------------------------|------------------------------------------------------------------------------------|
-| otel.metrics.exporter=prometheus | OTEL_METRICS_EXPORTER=prometheus | Select the Prometheus exporter                                                     |
-| otel.exporter.prometheus.port    | OTEL_EXPORTER_PROMETHEUS_PORT    | The local port used to bind the prometheus metric server. Default is `9464`.       |
-| otel.exporter.prometheus.host    | OTEL_EXPORTER_PROMETHEUS_HOST    | The local address used to bind the prometheus metric server. Default is `0.0.0.0`. |
+| System property                                         | Environment variable                                    | Description                                                                                                                          |
+|---------------------------------------------------------|---------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------|
+| otel.metrics.exporter=prometheus                        | OTEL_METRICS_EXPORTER=prometheus                        | Select the Prometheus exporter                                                                                                       |
+| otel.exporter.prometheus.port                           | OTEL_EXPORTER_PROMETHEUS_PORT                           | The local port used to bind the prometheus metric server. Default is `9464`.                                                         |
+| otel.exporter.prometheus.host                           | OTEL_EXPORTER_PROMETHEUS_HOST                           | The local address used to bind the prometheus metric server. Default is `0.0.0.0`.                                                   |
+| otel.exporter.prometheus.resource.attributes.add.regexp | OTEL_EXPORTER_PROMETHEUS_RESOURCE_ATTRIBUTES_ADD_REGEXP | If set, resource attributes whose key matches the given regular expression will be added to each data point label. Default is unset. |
```

Comment:
Ok. You're basically saying:
1. Remove the system property (and environment variables) code and documentation. 
2. Replace it with Predicate<String> for resource attribute key filtering.

The configuration PR I saw you are working on for adding support for this, will also include a ENV variable?

Reply:
Ok, I started changing it from regxp to Predicate<String> and I came to the last bit of removing the parsing of the system property:
```
    String resourceAttributesRegexp =
        config.getString("otel.exporter.prometheus.resource.attributes.add.regexp");
    if (resourceAttributesRegexp != null) {
      prometheusBuilder.setAllowedResourceAttributesFilter(resourceAttributesRegexp);
    }
```

then I suddenly realized. In Apache Pulsar I'm using AutoConfigred, since I want the user to configure it to suit them, but I need to be able to easily configure copying the `cluster`. How should this be done? Should I simply wait for the configuration repo to contain it?

Reply:
> Ok. You're basically saying:

Yes.

> The configuration PR I saw you are working on for adding support for this, will also include a ENV variable?

Seems unlikely.

I think as of today, waiting is the only option short of doing so your own programmatic configuration.

But a couple of things that may help:
- We could (should) extend [AutoConfigurationCustomizer](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/AutoConfigurationCustomizer.java) with a `addMetricReaderCustomier(BiFunction<? super MetricReader, ConfigProperties, ? extends MetricReader>)` to allow folks to perform programmatic customization of autoconfigured metric readers. 
- You can work with the spec to introduce the environment variable. Although this will be difficult with the [moratorium](https://github.com/open-telemetry/opentelemetry-specification/issues/2891#issuecomment-1289241503) on introducing new environment variables. Most of the activity is focussed around file configuration right now, which still has some missing pieces, but which is definitely going to have properties for things like adding resource attributes to the prometheus exporter.